### ISSUE TITLE
Bug: JavaScript Demo theme mode synchronised with computer theme mode and not mode toggle

### ISSUE THREAD

**Issue by bdrnglm** [mdn/yari#5381]
# Summary

When I toggle the theme of the website to dark or light mode it toggles correctly. But if I refresh the page while being in a mode opposite to the one of my computer the page loads with my selected mode but the JavaScript Demo window loads with the mode of my computer.

## Where did you spot the bug?

You can reproduce the bug in [this page](https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Global_Objects/Array/sort)  for instance

## What did you expect to happen?

The theme of the Demo should be bind to the website toggle and not the computer theme mode

## Steps to reproduce the issue

1. Set your computer theme to Dark Mode
2. Go on any page containing a JavaScript Demo windows (like [this one](https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Global_Objects/Array/sort)) and toggle the theme button to Light Mode. You should see the JavaScript demo window turn to light mode.
3. Refresh the page. You should see the whole page being in the Light Mode but the JavaScript Demo window set to Dark Mode

### Device

- [ x ] Desktop

### Browser

- [ x ] Edge
- [ x ] Safari

### Operating system

- [ x ] Windows: 10
- [ x ] Mac OS: 11

## Screenshots

![Screenshot 2022-03-01 at 19 48 43](https://user-images.githubusercontent.com/3275493/156230236-6d581aef-1e79-4052-935f-a8a47388d540.png)

**Comment by schalkneethling** [comment]
Will address this as part of https://github.com/mdn/yari/issues/5354 - Thank you for reporting @bdrnglm

**Comment by bdrnglm** [comment]
The problem is not completely fixed.
If you've set your page to dark mode by default and your OS mode is in light mode, the demo will be light while the website will be dark.

**Comment by schalkneethling** [comment]
Yeah, we were going to roll out the change but then backtracked on the implementation. Unfortunately the PR that reverted the changes did not re-open this issue. I am working on a new implementation today.
